### PR TITLE
Adding react-a11y and fixing known issues

### PR DESCRIPTION
**Pull request by jodiedoubleday**
#### What does this PR do? (please provide any background)
We really need to concentrate on creating beautiful user experience for everyone.
A11y is a measure of a computer system's accessibility to all people, including those with disabilities or impairments. React-a11y is a plugin which checks your components to see if they are compatible to a11y standards. It throws warning for everything that isn't.

#### What tests does this PR have?
None , its like linting your code

#### How can this be tested?
run `npm run docs` you should see no errors on the documentation page.
No to into the file `docs/src/customComponent.jsx` and change line 155 to remove the `role="button"` this make the page have poor accessibility.
You should now see an error thrown in the console (picture below)

#### Screenshots / Screencast
screen shot 2015-11-06 at 12 10 19

#### What gif best describes how you feel about this work?
50 cent

---
#### Developer Definition of Done/Quality Checklist (for PR author to complete BEFORE code review):
- [x] I have checked the [contributing document](../blob/master/CONTRIBUTING.md) and I'm happy for this to be reviewed.

#### Software Engineer or Developer review:
- [ ] I’ve witnessed the work behaving as expected (this could be on the authors machine or screencast).
- [ ] I’ve checked for coding anti-patterns.
- [ ] I’ve checked for appropriate test coverage.
- [ ] I’ve checked all the tests are passing.

#### Software Engineer or project guru review:
- [x] I’ve witnessed the work behaving as expected (this could be on the authors machine or screencast).
- [x] I’ve checked for coding anti-patterns.
- [x] I’ve checked for appropriate test coverage.
- [x] I’ve checked all the tests are passing.

### CODE REVIEW

File: docs/examples/BasketItem.jsx [hunk 1]
```diff
@@ -29,13 +29,13 @@ var example = (
...
       <UIToolkit.IconListItem icon="check">
-        <UIToolkit.BasketItem title={<a onClick={pretendLightBox.bind(null, '4th product')}>Fourth Product</a>} handleRemove={removeAThing.bind(null,'4th product')} price={100}>
+        <UIToolkit.BasketItem title={<a onClick={pretendLightBox.bind(null, '4th product')} role="link" tabIndex="0">Fourth Product</a>} handleRemove={removeAThing.bind(null,'4th product')} price={100}>
           Has a special title (could open a lightbox or something?)
         </UIToolkit.BasketItem>
       </UIToolkit.IconListItem>
 
       <UIToolkit.IconListItem icon="check">
-        <UIToolkit.BasketItem title={<a onClick={pretendLightBox.bind(null, '5th product')}>Fifth Product (no description)</a>} price={100} />
+        <UIToolkit.BasketItem title={<a onClick={pretendLightBox.bind(null, '5th product')} role="link" tabIndex="0">Fifth Product (no description)</a>} price={100} />
```
Comment (hxpaul) [inline docs/examples/BasketItem.jsx]: should these both have tab index 0?

**Comment by hxtomprice** [comment]
Dev

**Comment by hxtomprice** [comment]
I still get a few warnings, do these need fixing now?

**Comment by jodiedoubleday** [comment]
I believe tab index = 0 means stick with the natural tab index. so that should be okay

**Comment by jodiedoubleday** [comment]
Hey @hxtomprice what warning do you get? can you `rm -rf node_modules` and run `grunt build` before running `npm run docs`

**Comment by hxpaul** [comment]
is tab index 0 required if it means "follow the normal rules"?

**Comment by hxtomprice** [comment]
Run through those steps and am still getting a few on this screenshot http://grab.by/LTbO

**Comment by andrew-c** [comment]
Taking SE/Guru

**Comment by andrew-c** [comment]
Looks good :+1: I do get the same errors as Tom though: https://cloudup.com/c4n4iF7EEch

**Comment by hxtomprice** [comment]
@jodiedoubleday merge conflicts

**Comment by hxtomprice** [comment]
Getting some js errors with this branch which aren't occuring on master http://grab.by/M2by

**Comment by andrew-c** [comment]
I get the same errors as @hxtomprice

**Comment by andrew-c** [comment]
Yup :+1:

**Comment by hxtomprice** [comment]
:+1:

**Pull request merged**